### DISCUSSION TITLE
Loss function of the expectation value

### DISCUSSION THREAD

**Discussion by kdg930202** [netket/netket#1491] [Q&A]
This is a part of source code "netket/vqs/full_summ/expect.py"

I just don't understand "**ΔOΨ**".

I guess it is the loss function of the expectation value.

but cannot understand why it is defined like that **ΔOΨ = (OΨ - expval_O * Ψ).conj() * Ψ**.

    expval_O = (Ψ.conj() * OΨ).sum()
    variance = jnp.sum(jnp.abs(OΨ - expval_O * Ψ) ** 2)

    ΔOΨ = (OΨ - expval_O * Ψ).conj() * Ψ

    _, vjp_fun, *new_model_state = nkjax.vjp(
        lambda w: model_apply_fun({"params": w, **model_state}, σ, mutable=mutable),
        parameters,
        conjugate=True,
        has_aux=is_mutable,
    )

    Ō_grad = vjp_fun(ΔOΨ)[0]

    new_model_state = new_model_state[0] if is_mutable else None

    return (
        Stats(mean=expval_O, error_of_mean=0.0, variance=variance),
        Ō_grad,
        new_model_state,
    )

**Comment by PhilipVinc** [comment]
What is not clear? May I suggest that you write the formula that you were expecting and show what lines don't match it?
By the way, consider that `O` in those lines is not the $O_k(\sigma)=\partial_k\log\psi(\sigma)$ but it's the operator of which we are computing the expectation value, and `O_expval` is $\langle\psi|\hat{O}|\psi\rangle/\langle\psi|\psi\rangle$ that is the expectation value of it.
